pkg/kubelet/rkt/rkt.go

```diff
@@ -2103,20 +2102,28 @@ func (r *Runtime) PortForward(pod *kubecontainer.Pod, port uint16, stream io.Rea
 		}
 		return fmt.Errorf("more than one running rkt pod for the kubernetes pod [%s]", strings.Join(podlist, ", "))
 	}
+	listPod := listResp.Pods[0]
 
 	socatPath, lookupErr := exec.LookPath("socat")
 	if lookupErr != nil {
 		return fmt.Errorf("unable to do port forwarding: socat not found.")
 	}
 
-	args := []string{"-t", fmt.Sprintf("%d", listResp.Pods[0].Pid), "-n", socatPath, "-", fmt.Sprintf("TCP4:localhost:%d", port)}
-
-	nsenterPath, lookupErr := exec.LookPath("nsenter")
-	if lookupErr != nil {
-		return fmt.Errorf("unable to do port forwarding: nsenter not found.")
+	var args []string
+	var fwCaller string
+	if strings.Contains(r.config.Stage1Image, "kvm") {
```

Comment:
@yifan-gu As far as I see, I am unable to get pod manifest in this scope. All pod information I got is [kubecontainer.pod](https://godoc.org/k8s.io/kubernetes/pkg/kubelet/container#Pod), or at most [manifest from api-service](https://github.com/kubernetes/kubernetes/blob/release-1.4/pkg/kubelet/rkt/rkt.go#L598). Am I wrong?